### ISSUE TITLE
[Bug] Calculated padded input size per channel: (18). Kernel size: (25). Kernel size can't be greater than actual input size

### ISSUE THREAD

**Issue by bemnet4u** [coqui-ai/TTS#1270]
## 🐛 Description

Unable to train the speedyspeech model with new data set. Error `Calculated padded input size per channel: (18). Kernel size: (25). Kernel size can't be greater than actual input size`
The dataset I have has been used to train the VITS model successfully but its too slow for inference (20 seconds on 2 core CPU). I am exploring if this model will work better as I have read it result in a better inference model.  

### To Reproduce
python recipes/amharic/train_speedy_speech_amharic.py from https://github.com/bemnet4u/TTS/blob/amharic/recipes/amharic/train_speedy_speech_amharic.py 
or Notebook here https://colab.research.google.com/drive/1nZOjBIdJnXTswkRkXYupiXYFAtZePDul?usp=sharing

### Environment
```{
    "CUDA": {
        "GPU": [
            "Tesla P100-PCIE-16GB"
        ],
        "available": true,
        "version": "11.1"
    },
    "Packages": {
        "PyTorch_debug": false,
        "PyTorch_version": "1.10.0+cu111",
        "TTS": "0.5.0",
        "numpy": "1.21.5"
    },
    "System": {
        "OS": "Linux",
        "architecture": [
            "64bit",
            ""
        ],
        "processor": "x86_64",
        "python": "3.7.12",
        "version": "#1 SMP Tue Dec 7 09:58:10 PST 2021"
    }
}
```

### Additional context

There is the error I see. 
```
tensor([[[ 15.5020,  10.2680,  12.5079,  ...,  14.2169,  14.4178,  11.7207],
         [  1.9847,  -1.3257,  -5.8059,  ...,  -0.9510,  -2.7920,  -1.0401],
         [  0.5523,   3.4375,  -1.4630,  ...,  -0.0986,   3.4299,  -0.1949],
         ...,
         [-11.5210,  -7.4054,  -4.6783,  ...,  -7.2718,  -9.9058, -10.6706],
         [ -3.0461,   3.1941,   3.6398,  ...,  -0.0750,  -2.5927,   0.1630],
         [-14.9317, -11.9461,  -8.4180,  ..., -12.1159, -15.6248, -18.0321]]],
       device='cuda:0')
tensor([[[ 15.1302,  10.0421,  13.7486,  ...,  13.8451,  14.0460,  11.3489],
         [  1.6002,  -1.7102,  -5.8203,  ...,  -1.3355,  -3.1765,  -1.4246],
         [  0.2073,   3.0925,  -1.7967,  ...,   0.0757,   3.0849,  -0.5399],
         ...,
         [-12.0006,  -7.8850,  -5.1579,  ...,  -7.7514, -10.3853, -11.1502],
         [ -3.4630,   2.7772,   3.8240,  ...,  -0.2950,  -3.0095,  -0.2539],
         [-15.3527, -10.5941,  -8.5945,  ..., -12.5370, -16.0458, -18.4531]]],
       device='cuda:0')
tensor([[[ 14.7298,   9.6417,  13.3482,  ...,  13.4447,  13.6456,  10.9485],
         [  1.2190,  -2.0915,  -6.2015,  ...,  -1.7167,  -3.5577,  -1.8059],
         [ -0.2172,   2.6680,  -2.2212,  ...,  -0.3488,   2.6604,  -0.9644],
         ...,
         [-12.3904,  -8.2748,  -5.5477,  ...,  -8.1412, -10.7751, -11.5400],
         [ -3.8872,   2.3530,   3.3998,  ...,  -0.7192,  -3.4338,  -0.6781],
         [-15.7022, -10.9436,  -8.9440,  ..., -12.8864, -16.3953, -18.8026]]],
       device='cuda:0')
tensor([[[ 14.2813,   9.1932,  12.8997,  ...,  12.9962,  13.1971,  10.5000],
         [  0.8450,  -2.4655,  -6.5755,  ...,  -2.0907,  -3.9317,  -2.1799],
         [ -0.5719,   2.3134,  -2.5759,  ...,  -0.7035,   2.3057,  -1.3190],
         ...,
         [-12.7635,  -8.6479,  -5.9209,  ...,  -8.5143, -11.1483, -11.9131],
         [ -4.3099,   1.9304,   2.9772,  ...,  -1.1418,  -3.8564,  -1.1008],
         [-16.1298, -11.3712,  -9.3716,  ..., -13.3141, -16.8229, -19.2302]]],
       device='cuda:0')
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/TTS/trainer.py", line 1007, in fit
    self._fit()
  File "/usr/local/lib/python3.7/dist-packages/TTS/trainer.py", line 996, in _fit
    self.test_run()
  File "/usr/local/lib/python3.7/dist-packages/TTS/trainer.py", line 958, in test_run
    figures, audios = self.model.test_run(self.training_assets)
  File "/usr/local/lib/python3.7/dist-packages/TTS/tts/models/base_tts.py", line 424, in test_run
    do_trim_silence=False,
  File "/usr/local/lib/python3.7/dist-packages/TTS/tts/utils/synthesis.py", line 295, in synthesis
    outputs = run_model_torch(model, text_inputs, speaker_id, style_mel, d_vector=d_vector, language_id=language_id)
  File "/usr/local/lib/python3.7/dist-packages/TTS/tts/utils/synthesis.py", line 100, in run_model_torch
    "language_ids": language_id,
  File "/usr/local/lib/python3.7/dist-packages/torch/autograd/grad_mode.py", line 28, in decorate_context
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/TTS/tts/models/forward_tts.py", line 601, in inference
    o_de, attn = self._forward_decoder(o_en, o_dr, x_mask, y_lengths, g=None)
  File "/usr/local/lib/python3.7/dist-packages/TTS/tts/models/forward_tts.py", line 397, in _forward_decoder
    o_de = self.decoder(o_en_ex, y_mask, g=g)
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/TTS/tts/layers/feed_forward/decoder.py", line 229, in forward
    o = self.decoder(x, x_mask, g)
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/TTS/tts/layers/feed_forward/decoder.py", line 160, in forward
    o = self.res_conv_block(x, x_mask)
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/TTS/tts/layers/generic/res_conv_bn.py", line 125, in forward
    o = block(o)
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/TTS/tts/layers/generic/res_conv_bn.py", line 80, in forward
    return self.conv_bn_blocks(x)
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/container.py", line 141, in forward
    input = module(input)
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/TTS/tts/layers/generic/res_conv_bn.py", line 42, in forward
    o = self.conv1d(x)
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/conv.py", line 301, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/conv.py", line 298, in _conv_forward
    self.padding, self.dilation, self.groups)
RuntimeError: Calculated padded input size per channel: (18). Kernel size: (25). Kernel size can't be greater than actual input size

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/content/drive/MyDrive/VoiceCloning/TTS_amharic/recipes/amharic/train_speedy_speech_amharic.py", line 93, in <module>
    trainer.fit()
  File "/usr/local/lib/python3.7/dist-packages/TTS/trainer.py", line 1026, in fit
    remove_experiment_folder(self.output_path)
  File "/usr/local/lib/python3.7/dist-packages/TTS/utils/generic_utils.py", line 74, in remove_experiment_folder
    print(" ! Run is removed from {}".format(experiment_path))
  File "/usr/local/lib/python3.7/dist-packages/TTS/trainer.py", line 1184, in write
    with open(self.log_file, "a", encoding="utf-8") as f:
FileNotFoundError: [Errno 2] No such file or directory: '/content/drive/MyDrive/VoiceCloning/TTS_amharic/recipes/amharic/speedy_speech_amharic-February-20-2022_12+41AM-be4f23a5/trainer_0_log.txt'
```

**Comment by erogol** [comment]
It is not a bug. You have input samples that are shorter than the size of the convolutional layer. Try filtering short samples.

**Comment by arif334** [comment]
@bemnet4u Have you been able to train `speedy_speech`? In case you do, what changes have you made?

**Comment by candlewill** [comment]
> It is not a bug. You have input samples that are shorter than the size of the convolutional layer. Try filtering short samples.

Indeed, the input length should not be shorter than the convolution kernel size. 

However, sometimes we have actually short text at the training and inference stage. 

Does it mean SpeedySpeech is not appliable for short text synthesis?

**Comment by arif334** [comment]
> > @bemnet4u Have you been able to train `speedy_speech`? In case you do, what changes have you made?
> 
> You can see the linked fork for the change I made but I was not to get the model to converge after 25 hr. I was able to train vits model with much better quality in 10hr with the same dataset. But Inference was slow (5-9 sec on 4 vcpu vm) for vits that is why was looking at speedy_speech.

Thanks for the feedback.

**Comment by Edresson** [comment]
The long term fix for this issue was suggested in issue #1388 and probably will be fixed soon.
